selfdrive/hardware/tici/agnos.py

```diff
@@ -33,6 +40,13 @@ def read(self, length: int) -> bytes:
         break
       out = self.decompressor.decompress(compressed)
       self.buf += out
+      self.downloaded += len(out)
+      now = time.monotonic()
+      if (now - self.last_read > 1):
```

Comment:
this is measuring the speed in a pretty naive way since bytes are being read from the buffer in between chunks, but assuming the download takes up the majority of time, it could be somewhat accurate